### DISCUSSION TITLE
Shorthand for properties which get and set from the same place

### DISCUSSION THREAD

**Discussion by sab39** [dotnet/csharplang#5915] [Language Ideas] [answered]
I've found it to be pretty common to write properties that are effectively "aliases" for another expression. The *most* common case is the one that auto-properties are already shorthand for: the property is an alias for a hidden background field. But often that's not the case and the expression is more complex. For example,

```cs
interface IActivatable
{
  bool IsActive { get; set; }
}
interface IEnableable
{
  bool IsEnabled { get; set; }
}

class Widget : IActivatable, IEnableable
{
  public bool IsActive { get; set; }
  public bool IsEnabled { get => IsActive; set => IsActive = value; }

  public string LabelText { get => LabelControl.Text; set => LabelControl.Text = value; }
}
```

In simple cases like this it's not too bad, but if the expression being "aliased" is long or complicated, typing it out twice introduces opportunities for mistakes and makes it hard to see at a glance whether it's really the same in the getter and setter. It seems to me that this pattern is common enough to merit syntactic sugar. I semi-jokingly responded to an LDM Quote of the Day about "spaceship operators" proposing a syntax for this: `public bool IsEnabled <=> IsActive;`.

By itself this doesn't allow putting different accessibility modifiers on each accessor, or making the setter an `init`, so to account for that the syntax could be extended slightly: `public bool IsEnabled { get; private set; } <=> IsActive;`. The simpler form where the braces are admitted would then be a shorthand for `{ get; set; }`.

Further, since the proposed `<=>` syntax inherently implies both a getter and setter, it's not even necessary to type accessors that aren't being customized in some way, so the the previous example could be shortened to `public bool IsEnabled { private set; } <=> IsActive;` with the default `get;` being implied.

I'm pretty sure this syntax would be useful for many people, but I understand the starting-at-minus-100-points rule and that it may well not meet the bar to actually make it into the language. No harm in suggesting it though!

**Comment by svick** [comment, marked as answer]
This is a duplicate of https://github.com/dotnet/csharplang/discussions/1007.

**Reply by sab39** [reply]
Thanks! Subscribed to that now.
